infor-design/enterprise
# 4091 - Add truncated style to Breadcrumbs

**Explain the _details_ for making this change. What existing problem does the pull request solve?**
This PR:
- Adds a truncated style to the Breadcrumb component, per request from @gushanson
- Sets this new style to the default (old wrapping style is still optionally available / used as a fallback in IE11)
- Makes various visual tweaks to all Breadcrumb variants (changes in padding/spacing/etc)
- Adds a missing API endpoint (`get disabled() on individual breadcrumbs`)
- Adds/updates tests for the new truncation feature

**Related github/jira issue (required)**:
Closes #4091

**Steps necessary to review your pull request (required)**:
Pull this branch, build, run the demoapp. Then test all scenarios below.  On all scenarios, also all combinations of RTL, Uplift, Personalization (all should be working).

**NOTE:** The resize detection was created using a `ResizeObserver`.  This primitive doesn't exist in IE11, so if you test this feature in IE11, the breadcrumb list does not truncate.  However, it should gracefully degrade back to the original, non-truncated style in this case, so it's worth testing.

_Test basic samples:_
- Open http://localhost:4000/components/breadcrumb/example-index.html
- When this page loads at a large form-factor, no truncation should occur.
- Also open http://localhost:4000/components/breadcrumb/example-from-settings.html
- When this page loads at a large form-factor, no truncation should occur.
- Clicking Destroy/Invoke should work without causing any JS console errors.

_Test truncation:_
- Open http://localhost:4000/components/breadcrumb/example-add-remove.html
- Click the "Add" button to add enough breadcrumbs to the list to cause truncation.
- Click the "..." button that appears to the left of the breadcrumbs to open the overflow menu.  When it opens, it should display all list items that are cut off and/or hidden from the list.
- Click any of the items in the overflow menu.  When clicking, a toast should display with the correct Breadcrumb name (this signals that the Breadcrumb item's callback method is properly being triggered from the overflow menu).
- Try adding/removing breadcrumbs, adding disabled breadcrumbs, and using the `Tab`/`Shift`+`Tab` keys to navigate across the Breadcrumb list. Any items that are cut off/hidden/disabled should not become focused.  The Popupmenu should open using its normal keyboard handling.

_Test header styles:_
Try the following pages to test the standard Header breadcrumb style:
- http://localhost:4000/components/breadcrumb/example-navigation-breadcrumbs.html
- http://localhost:4000/components/breadcrumb/example-navigation-breadcrumbs.html?theme=uplift
http://localhost:4000/components/breadcrumb/example-navigation-breadcrumbs.html?theme=uplift&variant=dark&colors=7928E1

Try the following pages to test the alternate Header breadcrumb style:
- http://localhost:4000/components/breadcrumb/example-navigation-breadcrumbs-alternate.html
- http://localhost:4000/components/breadcrumb/example-navigation-breadcrumbs-alternate.html?theme=uplift
http://localhost:4000/components/breadcrumb/example-navigation-breadcrumbs-alternate.html?theme=uplift&variant=dark&colors=7928E1

Try various combinations of themes/personalization.

_Test where `truncate: false` is set:_
- Open http://localhost:4000/components/breadcrumb/example-no-truncate.html
- Observe this breadcrumb list will always wrap and never truncate.

**Included in this Pull Request**:
